tutorhastexo/templates/hastexo/apps/hastexo/settings/settings.py

```diff
@@ -0,0 +1,63 @@
+import json
+import os
+
+from distutils.util import strtobool
+
+
+BASE_DIR = os.path.dirname(os.path.dirname(os.path.dirname(__file__)))
+
+SECRET_KEY = "{{ HASTEXO_SECRET_KEY }}"
+DEBUG = bool(strtobool("{{ HASTEXO_DEBUG }}"))
+LOGGING = {
+    'version': 1,
+    'disable_existing_loggers': False,
+    'formatters': {
+        'default': {
+            'format': '%(levelname)s:%(name)s:%(message)s',
+        },
+    },
+    'handlers': {
+        'console': {
+            'class': 'logging.StreamHandler',
+            'formatter': 'default',
+        },
+    },
+    'root': {
+        'handlers': ['console'],
+        'level': 'DEBUG' if DEBUG else 'WARNING'
+    },
+}
+
+ALLOWED_HOSTS = ''
+
+DATABASES = {
+    'default': {
+        'ENGINE': 'django.db.backends.mysql',
+        'NAME': "{{ OPENEDX_MYSQL_DATABASE }}",
+        'USER': "{{ OPENEDX_MYSQL_USERNAME }}",
+        'PASSWORD': "{{ OPENEDX_MYSQL_PASSWORD }}",
+        'HOST': "{{ MYSQL_HOST }}",
+        'PORT': "{{ MYSQL_PORT }}",
+        'ATOMIC_REQUESTS': False,
+        'CONN_MAX_AGE': 0,
+        'OPTIONS': {},
+    }
+}
+
+INSTALLED_APPS = [
+    'django.contrib.admin',
+    'django.contrib.auth',
+    'django.contrib.contenttypes',
+    'django.contrib.sessions',
+    'django.contrib.messages',
+    'django.contrib.staticfiles',
+    'channels',
+    'hastexo_guacamole_client'
+]
+
+ASGI_APPLICATION = 'hastexo_guacamole_client.routing:application'
+
+XBLOCK_SETTINGS = {
+    "hastexo": json.loads(
```

Comment:
Do you really need to convert from a Python dictionary (which is what `HASTEXO_XBLOCK_SETTINGS` should be in the Jinja2 rendering context) to JSON with the `tojson()` filter, only to convert right back to a dictionary with `json.loads()`? Can't you just use the dictionary directly?

Reply:
I spent so much time on this and couldn't figure out how this should work, somehow it always ended up being one long string instead of a dict. Eventually, I looked for examples and saw this: https://github.com/overhangio/tutor-ecommerce/blob/master/tutorecommerce/templates/ecommerce/apps/ecommerce/settings/partials/common.py#L80 So I tried that and it finally worked.